pkg/core/proxy/ca.go

```diff
@@ -91,6 +91,15 @@ func getCaPaths() ([]string, error) {
 
 // to extract ca certificate to temp
 func extractCertToTemp() (string, error) {
+
+	if err := utils.SetUmask(0); err != nil {
+		return "failed to set Umask", err
+	}
+
+	if err := utils.SetUmask(0022); err != nil {
```

Comment:
You have to make sure that the umask is set back to its before value which can be (027, etc). So, need to first extract the before umask of system.

Reply:
And here setUmask should be after CreateTemp method.

Reply:
Your requested changes have been made